### ISSUE TITLE
Remember alphabet layout per App: off not working

### ISSUE THREAD

**Issue by archont00** [AnySoftKeyboard/AnySoftKeyboard#2970]
### Steps to reproduce
1. Install & enable multiple keyboard layouts (languages) - I use English QWERTZ and Czech QWERTZ.
2. Go to `Settings // Language // Tweaks and More // Remember alphabet layout per App`  and disable it.
3. Use keyboard in multiple apps.

### Expected behaviour

ASK should remember and keep the last language keyboard across any app until the user actually changes it.

### Actual behaviour

ASK kind of randomly switches to English layout whenever the user switches between apps or even between input fields of different type (`URL`, `Username`, ...) within the same app. 

It does not help to change the order of languages in Settings - ASK always re-sets to English (QWERTZ in my case).

**Android OS version:** 11

**Device manufacturer and model:** Xiaomi Mi 10 Pro (Global) with XIAOMI.EU ROM 12.6

**List of installed add-ons (like languages, or themes):** `Czech for AnySofKeyboard`

**Comment by pokulo** [comment]
Experienced a similar behaviour. But for me it seems to be connected to only some apps. A bit more of testing showed, that switching the type of input field even within the same app will automatically switch layout and language. So using a messanger-Text input and the typing a URL into the browser-app will imediately reset languagge to english.
Even switching per-app-remember on is not helpful for example in Contact-Edit view, where there are, mail, text, phone,date fields all together.
The keybord should alsways remember the language regardless of the type of input field.

**Comment by pokulo** [comment]
I am using LineageOS 18.1 mith microg on Xiaomi Mi6 (sagit).

**Comment by pokulo** [comment]
I found a config to disable this behaviour at least for URLs and EMail-Fields under:
Language->Tweaks and more->Tweaks->Layout for URL fields
This explains that behaviour. Maybe this Feature could be repaired so the language is kept unchanged at least when switching back to non-URL fields.
Does this help you @archont00 ?

**Comment by Soundtoxin** [comment]
I'm getting forced into a url layout mode in my browser no matter what settings I change. It's messing up my bottom row, and afaict there is no apostrophe key anywhere when this happens.

**Comment by menny** [comment]
UI->Tweaks and more->Enabled specialized input-field modes. Uncheck the "URL input field".

**Comment by Soundtoxin** [comment]
> UI->Tweaks and more->Enabled specialized input-field modes. Uncheck the "URL input field".

Oh my god. Thank you. Worked. Man, the settings are split up way too much in this app. I wish it were all on one screen and/or had a search.

**Comment by GitHorst** [comment]
> UI->Tweaks and more->Enabled specialized input-field modes. Uncheck the "URL input field".

this does not solve the original issue, does it?

what I observe in addition to OP: it even switches back to english when changing screen orientation! --> start typing (e.g. in german), then rotate the phone: the layout is english

**Comment by GitHorst** [comment]
@archont00 can you test whether the suggestion by @menny works for you? for me it does not for english + german keyboard. sometimes it does remember the last layout set by the user, sometimes it does not. I will also create a new bug report for the screen orientation resetting the layout...

**Comment by GitHorst** [comment]
@menny the layout reset upon screen rotation has already been reported: e.g. #713 and 592

should I comment there to reopen or create a new ticket?

**Comment by menny** [comment]
Comment in the other issue.

Also, I can close this ticket?

**Comment by pokulo** [comment]
> Also, I can close this ticket?

Turning URL-Layout off, only because of usage of multiple languages, is not a proper solution to the original bug report. The manually chosen language should persist over all layouts until switched manually.

**Comment by GitHorst** [comment]
> Also, I can close this ticket?

I agree w/ @pokulo and in addition: for me this workaround did not do anything. As I mentioned above, ASK still "randomly"(?) switches back to english (also w/o screen orientation changes in-between!) after switching between a couple of apps w/ text-fields.
